src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -282,6 +354,197 @@ private boolean validatePasswords(String identityStorePass,
+
+        if (isPropValueValidated(legacyProvidersString)) {
+            isLegacyProvidersExists = true;
+            String[] legacyProviders = addStringToArray(legacyProvidersString);
+            addToProvidersMap(legacyProviders, PROP_SECRET_REPOSITORIES);
+        }
+    }
+
+    /**
+     * Read the providers listed under secretProviders property.
+     *
+     * @param secretConfigurationProperties All the configuration properties.
+     */
+    private void readNovelProviders(Properties secretConfigurationProperties) {
+
+        String novelProvidersString =
+                getPropertiesFromSecretConfigurations(secretConfigurationProperties, PROP_SECRET_PROVIDERS);
+
+        if (isPropValueValidated(novelProvidersString)) {
+            isNovelProvidersExists = true;
+            String[] novelProviders = addStringToArray(novelProvidersString);
+            addToProvidersMap(novelProviders, PROP_SECRET_PROVIDERS);
+        }
+    }
+
+    /**
+     * Terminates if either properties, secretRepositories or secretProviders haven`t been configured.
+     */
+    private void validateSecureVaultStatus() {
+
+        if (!(isLegacyProvidersExists || isNovelProvidersExists)) {
+            if (log.isDebugEnabled()) {
+                log.debug("No secret repositories have been configured.");
+            }
+            return;
+        }
+    }
+
+    /**
+     * Util method to add all the providers from providers array to providers hash map along with the
+     * type (secretRepositories or secretProviders).
+     *
+     * @param providersArray Repositories array and secretProviders array generated from the properties,
+     *                       secretRepositories or secretProviders.
+     * @param providerType   SecretRepositories or secretProviders.
+     */
+    private void addToProvidersMap(String[] providersArray, String providerType) {
+
+        for (String providerName : providersArray) {
+            providers.put(providerName, providerType);
+        }
+    }
+
+    /**
+     * Util method for getting property values from the secret-conf file.
+     *
+     * @param secretConfigProps All the properties under secret configuration file.
+     * @param propName          Name of the property.
+     * @return Returns the value for the give property.
+     */
+    private static String getPropertiesFromSecretConfigurations(Properties secretConfigProps, String propName) {
+
+        return MiscellaneousUtil.getProperty(secretConfigProps, propName, null);
+    }
+
+    /**
+     * Validate the property value to avoid the processing of null values.
+     *
+     * @param propValue Value of the required property.
+     * @return Return true if not null.
+     */
+    private static boolean isPropValueValidated(String propValue) {
+
+        if (isEmpty(propValue)) {
+            if (log.isDebugEnabled()) {
+                log.debug("No secret repository has been configured.");
```

Comment:
If this case prints a log, we should better print a log for the true case as well.

Reply:
Added a debug log for the true case